billy/app.py

```diff
@@ -6,27 +6,42 @@
 
 from loguru import logger
 
+import secrets
 import zipfile
 import io
 from requests import HTTPError, ConnectionError
 from http import HTTPStatus
 
-from flask import Flask, request, jsonify, make_response, send_file, g
+from flask import Flask, request, jsonify, make_response, redirect, send_file, g, url_for, make_response
 from flask_marshmallow import Marshmallow
+from flask_dance.consumer import OAuth2ConsumerBlueprint
 from marshmallow import fields
 from flask_marshmallow.sqla import HyperlinkRelated
 from marshmallow_sqlalchemy import SQLAlchemyAutoSchema, SQLAlchemySchema, auto_field
-
 from flask_mail import Mail, email_dispatched
 from smtplib import SMTPException
 from flask_cors import CORS
 
+HITOBITO_BASE = os.getenv('HITOBITO_HOST')
+UNPROTECTED_PATH = '/oauth'
+
+dance = OAuth2ConsumerBlueprint(
+    "billy", __name__,
+    client_id=os.getenv('HITOBITO_OAUTH_CLIENT_ID'),
+    client_secret=os.getenv('HITOBITO_OAUTH_SECRET'),
+    base_url=HITOBITO_BASE,
+    token_url="{base}/oauth/token".format(base=HITOBITO_BASE),
+    authorization_url="{base}/oauth/authorize".format(base=HITOBITO_BASE),
+    redirect_url=os.getenv('REDIRECT_URL_LOGIN'),
+    scope=['email', 'name', 'with_roles', 'api']
+)
+
 # init
 app = Flask(__name__)
+app.secret_key = secrets.token_urlsafe(32)
+app.register_blueprint(dance, url_prefix=UNPROTECTED_PATH)
 CORS(app)
 
-load_dotenv('./env/mail.env')
```

Comment:
This should either remain until  a better solution is implemented, or the README updated.